### PR TITLE
Added formatting

### PR DESCRIPTION
**Pull request by Nicholas-Baron**
A basic `.clang-format` for this template.

### CODE REVIEW

**Review by JamesGaier — COMMENTED** [review]
LGTM

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+AlignOperands: AlignAfterOperator
+AlignTrailingComments: true
+AlignAfterOpenBracket: Align
+
+# Allowed shortenings
+AllowShortBlocksOnASingleLine: true
+AllowShortFunctionsOnASingleLine: All
+AllowShortIfStatementsOnASingleLine: Always
+AllowShortLoopsOnASingleLine: true
+BreakBeforeBraces: Attach
```
Comment (JamesGaier) [inline .clang-format]: Is this Allman style or 1TBS?

Comment (Nicholas-Baron) [reply .clang-format]: This is 1TBS.

Comment (JamesGaier) [reply .clang-format]: Can you switch to Allman?

Comment (JamesGaier) [reply .clang-format]: or not enforce curly style?

Comment (Nicholas-Baron) [reply .clang-format]: Now using clang-format's built-in Allman style.

**Comment by JamesGaier** [comment]
Looks good.

**Pull request merged**